main.go

```diff
@@ -0,0 +1,280 @@
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/bitrise-io/go-utils/pathutil"
+
+	"github.com/hashicorp/go-version"
+
+	"github.com/bitrise-io/bitrise/tools/timeoutcmd"
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+	"github.com/kballard/go-shellquote"
+)
+
+const (
+	platformAndroid   = "android"
+	platformIos       = "ios"
+	credentialPathFmt = "%s/.config/gcloud/application_default_credentials.json"
+	baseURL           = "https://github.com/TestArmada/flank"
+	resultsDir        = "./results"
+)
+
+type config struct {
+	ServiceAccountJSON stepconf.Secret `env:"google_service_account_json,required"`
+	ConfigPath         string          `env:"config_path,file"`
+	Version            string          `env:"version,required"`
+	CommandFlags       string          `env:"command_flags"`
+}
+
+// returns android if there is an app field under gcloud in the config yml
+func detectPlatform(configYMLPath string) (string, error) {
+	var androidApp struct {
+		Gcloud struct {
+			App string `yaml:"app"`
+		} `yaml:"gcloud"`
+	}
+
+	ymlBytes, err := fileutil.ReadBytesFromFile(configYMLPath)
+	if err != nil {
+		return "", err
+	}
+
+	if err := yaml.Unmarshal(ymlBytes, &androidApp); err != nil {
+		return "", err
+	}
+
+	if len(androidApp.Gcloud.App) > 0 {
+		return platformAndroid, nil
+	}
+	return platformIos, nil
+}
+
+// stores string under the default gcloud credential's local path
+func storeCredentials(cred string) error {
+	pth := fmt.Sprintf(credentialPathFmt, os.Getenv("HOME"))
+	if err := os.MkdirAll(filepath.Dir(pth), 0777); err != nil {
+		return err
+	}
+	return fileutil.WriteStringToFile(pth, cred)
+}
+
+// gets the tags list, splits the lines per tab and finds the prefix-truncated version strings
+// if the version string is a valid semver version then this function returns the latest one
+func getLatestVersion() (string, error) {
+	cmd := command.New("git", "ls-remote", "--tags", "--quiet", "https://github.com/TestArmada/flank")
+	out, err := cmd.RunAndReturnTrimmedCombinedOutput()
+	if err != nil {
+		return "", err
+	}
+
+	// hashes and version refs are separated by tab, output will be something like:
+	// a85cdf850b7562d38559b84d405c97f7d194603s refs/tags/flank_snapshot
+	// f0d902fdd947e7f2db43213929bdc6069a681624	refs/tags/v1.4.2
+	// 4f00292267eaa2b230c4a5705a50c5302d7f9dc1	refs/tags/v1.5.0
+	// 44ed2076a684a740c41660ab80052b4218633538	refs/tags/v1.6.0
+	// 65762d7888a8a6e7490cbe7433a725201cbd5304	refs/tags/v1.7.0
+	// eae4b1e50e7eb05c02fc678994c6425bf14e1701	refs/tags/v2.0.0
+
+	var versions []string
+	for _, line := range strings.Split(out, "\n") {
+		if ref := strings.Split(line, "\t"); len(ref) == 2 {
+			versions = append(versions, strings.TrimPrefix(ref[1], "refs/tags/"))
+		}
+	}
+
+	lastVersion, err := version.NewVersion("0.0.0")
```

Comment:
is `0.0.0` an existing version, what if no other version would found?

Reply:
It is not valid semver version tough.